framework/test/driver/BundleGetSymbolTest.cpp

```diff
@@ -0,0 +1,119 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "cppmicroservices/Bundle.h"
+#include "cppmicroservices/BundleActivator.h"
+#include "cppmicroservices/BundleContext.h"
+#include "cppmicroservices/Framework.h"
+#include "cppmicroservices/FrameworkFactory.h"
+#include "cppmicroservices/GetBundleContext.h"
+
+#include "cppmicroservices/SharedLibrary.h"
+
+#include "TestingConfig.h"
+#include "TestingMacros.h"
+#include "TestUtils.h"
+
+#include <stdexcept>
+
+using namespace cppmicroservices;
+
+void TestGetSymbolInvalidBundleInput()
+{
+    // Test for invalid bundle
+    
+    Bundle b;
+    US_TEST_FOR_EXCEPTION(std::invalid_argument , b.GetSymbol(nullptr,""));
+}
+
+void TestGetSymbolInvalidInput()
```

Comment:
The names should perhaps reflect the conditions that are being violated. 

e.g invalid symbol name, null shared library.

In the null shared library case, you won't need to even load the Shared Library